File: apps/api/internal/domain/supplier.go
package domain

import "time"

// Supplier is a company medicines are purchased from.
type Supplier struct {
	ID          string
	Name        string
	ContactName *string
	Email       *string
	Phone       *string
	Address     *string
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

File: apps/app/src/types/supplier.ts
export interface Supplier {
  id: string
  name: string
  contact_name: string | null
  email: string | null
  phone: string | null
  address: string | null
  created_at: string
  updated_at: string
}

export interface SupplierListResponse {
  data: Supplier[]
  total: number
  limit: number
  offset: number
}

export interface CreateSupplierInput {
  name: string
  contact_name: string | null
  email: string | null
  phone: string | null
  address: string | null
}

export interface UpdateSupplierInput {
  name: string
  contact_name: string | null
  email: string | null
  phone: string | null
  address: string | null
}
